### ISSUE TITLE
Use URLs instead of paths

### ISSUE THREAD

**Issue by wooorm** [vfile/vfile#86]
### Initial checklist

- [X] I read the [support docs](https://github.com/vfile/.github/blob/main/support.md)
- [X] I read the [contributing guide](https://github.com/vfile/.github/blob/main/contributing.md)
- [X] I agree to follow the [code of conduct](https://github.com/vfile/.github/blob/main/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Avfile&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

It would be nice to go all in on a format that is supported everywhere: URLs.
Instead of a platform dependent (unix vs windows) string.

However, URLS are *broader* than what we support, what should we do when URLs point to `data:`? `https://`? Etc?
We could change `to-vfile` -> `vfile-fs` and add a `vfile-fetch` too?

Also, URLs as strings are problematic: passing `url.href` to `fs.writeFile` will write a `file/Users/User/whatever/` folder in CWD. This could be confusing to users.
This means our API has to likely return URL objects instead of strings. But that has again the problem of being either slow or a reference.

It’s a bit harder with URLs to change folders or extensions. Our current getter/setters are a nice alternative.
We could drop those getters/setters, maybe go with some utilities exposed from here to get/set stuff?

Extensions are typically meaningless in URLs anyway. Perhaps we’d need to have mimetypes?

Maybe we could first inspire from, and later depend on / add onto https://developer.mozilla.org/en-US/docs/Web/API/File?

### Solution

?

### Alternatives

Many!

**Comment by wooorm** [comment]
Closing as this is still a bit vague; URLs have some benefits, strings have some benefits. I’d welcome more ideas on how to do this, and do this fast, plus thoughts on how the ecosystem should deal with `data:` or `https:`!

**Comment by JounQin** [comment]
When passing `path` or `url` to `new VFile`, it seems only setting `path` but not reading `path` and then set `value`, is this expected? It results `value` empty which is very unexpected, maybe.

**Comment by wooorm** [comment]
>  It results value empty which is very unexpected, maybe.

No, it is expected: this core package is for *virtual* files, they don’t have to exist on *this* hard drive, they could exist on some other hard drive. Or not at all.

If you want `fs` integration, see https://github.com/vfile/to-vfile. `read(new URL(...))` *does* read the file from disk
